src/SeleniumLibrary/keywords/screenshot.py

```diff
@@ -235,3 +237,20 @@ def _embed_to_log_as_file(self, path, width):
             f'<a href="{src}"><img src="{src}" width="{width}px"></a>',
             html=True,
         )
+    
+    @keyword
+    # def print_page_as_pdf(self, print_options: Optional[PrintOptions]=None,):
+    # def print_page_as_pdf(self, print_options: Optional[Union[PrintOptions, dict]]=None):
+    def print_page_as_pdf(self, print_options: Union[PrintOptions, dict, None]=None):
+        """ Print the current page as a PDF
+
+        """
+        if not print_options:
+            print_options = PrintOptions()
+            print_options.page_ranges = ['-']
+
+        base64code  = self.driver.print_page(print_options)
+        pdfdata = b64decode(base64code)
+        with open('test.pdf', mode='wb') as pdf:
```

Comment:
This is very preliminary. For example, here I just use a test file. I expect a fair more code and allowing the user to specify the optional filename or filename syntax when printing page as pdf.

Reply:
Would it make sense to be able to set a default set of PrintOptions? Meaning the user can set the PrintOptions once, and those settings will be used every time the print_page_as_pdf will be called?
Something like with setting the timeouts  
`${old_value}  Set Print to PDF options  scale=0.5 password=${pdfpassword}`
or something like that

Reply:
It does makes sense to have a context for print options. For the initial release I am going to leave that out though.